auth/auth.go

```diff
@@ -0,0 +1,21 @@
+package auth
+
+// Provider is an SSPI compatible authentication provider
+type Provider interface {
+	// GetAuth is responsible for returning an instance of the required Auth interface
+	GetAuth(user, password, service, workstation string) (Auth, bool)
+}
+
+// Auth is the interface for SSPI Login Authentication providers
+type Auth interface {
```

Comment:
is `Auth` perhaps too generic a name? The driver implements AAD auth using azidentity now, plus there's "sql auth". Maybe `WinAuth` and `SetWinAuthProvider` for this?

Reply:
Other drivers like ODBC and ado.net also refer to it as "Integrated" or "Trusted"

Reply:
IntegratedAuthenticationProvider
IntegratedAuthenticator
would both signal, to me at least, their intent.